### ISSUE TITLE
Getting Reference Fasta Files To Match GTF

### ISSUE THREAD

**Issue by brendanwee** [alexdobin/STAR#399]
When renaming the fasta header lines to match the GTF files what information should be retained? For example, I was able to correctly index a single chromosome by renaming my fasta header lines to chr11. However there were two reads, each with position indicators. Is this information lost? How should I format my fasta header line if my GTF file looks like this?

chr11	mm10_gold	exon	3100001	34545122	0.000000	+	.	gene_id "GL456157.2"; transcript_id "GL456157.2";
chr11	mm10_gold	exon	34681621	121982543	0.000000	+	.	gene_id "GL456158.2"; transcript_id "GL456158.2";

**Comment by bweestyle** [comment]
I am trying to use only a single chromosome for alignment so that I can do it on my personal computer. I have been downloading my reference data from the UCSC table browser. I can't seem to find the correct files to download.

Is the fasta file supposed to be a genome assembly? Or is it supposed to be the fasta files of Genes and Gene Predictions?

**Comment by alexdobin** [comment]
Hi @bweestyle 

I do not recommend mapping to portions of a genome at a time, as this will be much slower than mapping to the whole genome. How much RAM do you have on your laptop? For the mouse genome, you can fit it into 16GB, and possibly even into 8GB.

Cheers
Alex

**Comment by bweestyle** [comment]
Hey Alex, I tried the whole genome again and got a killed 9 error. from another thread I found, this is probably due to memory? I have 16G on this computer and I restarted right before running the command.

My command was Brendans-MacBook-Pro-3:alignment Brendan$ STAR --runThreadN 7 --runMode genomeGenerate --genomeDir genome/ --genomeFastaFiles Mus_musculus.GRCm38.75.dna.primary_assembly.fa --sjdbGTFfile Mus_musculus.GRCm38.75.gtf

When I open top it says I have 11G unused mem. 

Is there something I can do to sidestep this error?

THank you so much for the help

**Comment by bweestyle** [comment]
Here is my log file
[Log.out.txt](https://github.com/alexdobin/STAR/files/1910053/Log.out.txt)

**Comment by alexdobin** [comment]
Hi Brendan,

please try to add --genomeSAsparseD 2 i.e.
$ STAR --runThreadN 7 --runMode genomeGenerate --genomeDir genome/ --genomeFastaFiles Mus_musculus.GRCm38.75.dna.primary_assembly.fa --sjdbGTFfile Mus_musculus.GRCm38.75.gtf --genomeSAsparseD 2 --limitGenomeGenerateRAM 14000000000

This should fit the genome into 16GB of RAM.
Cheers
Alex

**Comment by bweestyle** [comment]
Hi Alex,

Thanks so much for responding on this. I am still getting the same problem. I ran this command:
$ STAR --runThreadN 7 --runMode genomeGenerate --genomeDir genome/ --genomeFastaFiles Mus_musculus.GRCm38.75.dna.primary_assembly.fa --sjdbGTFfile Mus_musculus.GRCm38.75.gtf --genomeSAsparseD 2 --limitGenomeGenerateRAM 14000000000

here is my log file:

[Log.out.txt](https://github.com/alexdobin/STAR/files/1921966/Log.out.txt)

I am going to try and run this on amazon web services and see if that works. please let me know if you can come up with a fix! 

Thanks so much!

**Comment by alexdobin** [comment]
Hi Brendan,

I think you are getting close to fitting the genome in 16GB.
please try to add --limitSjdbInsertNsj 250000  --genomeSAindexNbases 12 at the genome generation step.

Cheers
Alex

**Comment by hlhorsnell** [comment]
Hi everyone,

I am having a similar problem, for some reason the FASTA and GTF files downloaded from ENSEMBL are not compatible?!

./STAR --runThreadN 1 --runMode genomeGenerate --genomeDir /Users/harryhorsnell/Desktop/genomeDir --genomeFastaFiles Mus_musculus.GRCm38.cdna.all.fa --sjdbGTFfile Mus_musculus.GRCm38.92.gtf --limitGenomeGenerateRAM 72152541226 

Apr 26 13:49:22 ..... started STAR run
Apr 26 13:49:22 ... starting to generate Genome files
Apr 26 13:52:30 ... starting to sort Suffix Array. This may take a long time...
Apr 26 13:54:23 ... sorting Suffix Array chunks and saving them to disk...
Apr 26 14:03:00 ... loading chunks from disk, packing SA...
Apr 26 14:04:05 ... finished generating suffix array
Apr 26 14:04:05 ... generating Suffix Array index
Apr 26 14:05:19 ... completed Suffix Array index
Apr 26 14:05:19 ..... processing annotations GTF

Fatal INPUT FILE error, no valid exon lines in the GTF file: Mus_musculus.GRCm38.92.gtf
Solution: check the formatting of the GTF file. Most likely cause is the difference in chromosome naming between GTF and FASTA file.

Apr 26 14:05:34 ...... FATAL ERROR, exiting

I then tried to use the data downloaded from GENCODE but get the Killed 9 error

./STAR --runThreadN 1 --runMode genomeGenerate --genomeDir /Users/harryhorsnell/Desktop/genomeDir --genomeFastaFiles gencode.vM17.transcripts.fa --sjdbGTFfile gencode.vM17.chr_patch_hapl_scaff.annotation.gtf --limitGenomeGenerateRAM 72152541226 

Apr 26 14:05:58 ..... started STAR run
Apr 26 14:05:58 ... starting to generate Genome files
Killed: 9

I have 8GB of RAM on my Mac, Any suggestions?

Many thanks,
Harry.

**Comment by alexdobin** [comment]
Hi Harry,

the FASTA file you need for mapping to the whole genome is:
ftp://ftp.ensembl.org/pub/release-92/fasta/mus_musculus/dna/Mus_musculus.GRCm38.dna.primary_assembly.fa.gz
This should work fine with the Ensembl GTF.

Cheers
Alex

**Comment by bweestyle** [comment]
Hi Alex,

So sorry for the delay. I pushed ahead in our project using AWS. We've only just managed to walk through our pipeline for the first time. Ideally though it would be great if we could run it on our laptops so I've just revisited this.
It seemed like it moved through the steps much faster this time!
Unfortunately I am still getting the same error message. Below is my command and log file. If you have any more suggestions I would really appreciate it, but if this is growing old that's fine we can make do with a server!

STAR --runThreadN 7 --runMode genomeGenerate --genomeDir genome/ --genomeFastaFiles Mus_musculus.GRCm38.75.dna.primary_assembly.fa --sjdbGTFfile Mus_musculus.GRCm38.75.gtf --genomeSAsparseD 2 --limitGenomeGenerateRAM 14000000000 --limitSjdbInsertNsj 250000 --genomeSAindexNbases 12

[Log.out.txt](https://github.com/alexdobin/STAR/files/1953501/Log.out.txt)

StdOut:
Apr 26 20:52:38 ..... started STAR run
Apr 26 20:52:38 ... starting to generate Genome files
Apr 26 20:53:30 ... starting to sort Suffix Array. This may take a long time...
Apr 26 20:53:37 ... sorting Suffix Array chunks and saving them to disk...
Apr 26 21:05:03 ... loading chunks from disk, packing SA...
Apr 26 21:06:19 ... finished generating suffix array
Apr 26 21:06:19 ... generating Suffix Array index
Apr 26 21:08:05 ... completed Suffix Array index
Apr 26 21:08:05 ..... processing annotations GTF
Apr 26 21:08:15 ..... inserting junctions into the genome indices
Killed: 9

Thanks!
Brendan
